comp/core/config/config.go

```diff
@@ -92,6 +98,22 @@ func newConfig(deps dependencies) (Component, error) {
 		}
 	}
 
+	configRefreshInterval := config.GetDuration("agent_ipc.config_refresh_interval")
+	agentIPCPort := config.GetInt("agent_ipc.port")
+	if flavor.GetFlavor() != flavor.DefaultAgent &&
+		agentIPCPort > 0 &&
+		configRefreshInterval > 0 {
+
+		agentIPCHost := config.GetString("agent_ipc.host")
+		url := &url.URL{
+			Scheme: "https",
+			Host:   net.JoinHostPort(agentIPCHost, strconv.Itoa(agentIPCPort)),
+			Path:   "/config/v1/",
+		}
+		//TODO: use an actual context
+		go syncConfigWithCoreAgent(context.TODO(), config, url, configRefreshInterval)
```

Comment:
WDYT about using fx.Lyfecycle here ?

Reply:
Yes that's exactly what I planned to do but I need to read the doc to understand how to do it 😄